### DISCUSSION TITLE
Add ability to hide effects on preview

### DISCUSSION THREAD

**Discussion by AntonReality** [HaishinKit/HaishinKit.swift#1530] [Idea]
Is there plans to add ability to hide effects on preview? Currently in our app we need to add this functionality and I’m thinking of if you’re planing to do this. It was there previously [added](https://github.com/shogo4405/HaishinKit.swift/pull/1377) by @leo150 but it was removed after moving to new screen logic.
If not, I can develop it and create a PR.
Thoughts?

**Comment by shogo4405** [comment]
No, it doesn't.

You can display the raw CMSampleBuffer captured from the camera as follows:
https://github.com/shogo4405/HaishinKit.swift/discussions/1504#discussioncomment-9989140

If this doesn't satisfy you, please provide more details so I can consider it further.

**Reply by AntonReality** [reply]
Hey @shogo4405, thanks for the reply! This is not working in the example iOS project.

> func stream(_ stream: HaishinKit.IOStream, track: UInt8, didInput buffer: CMSampleBuffer) {
     print("BUFFER: \(buffer)")
}

This delegate method is never called if attachStream is not called.

Delegate is set:
`stream.delegate = self`.

Or did I understand something wrong?

**Reply by shogo4405** [reply]
What I want to know is whether the CMSampleBuffer obtained in didInput buffer meets your display requirements.
I thought it would be good to try using an MTHKView with 1px:1px attached to the stream.

If it meets the requirements, I plan to redesign it so that the camera can capture without attachStream.

**Reply by AntonReality** [reply]
How can I test that if I can obtain CMSampleBuffer from the delegate only if stream is attached?

**Reply by shogo4405** [reply]
I couldn’t understand the question! It would be helpful if you could tell me what you expect from the program and where you are stuck.

To answer your initial question, I only respond to implementation plans for sponsors and contributors. Pull requests are welcome, but not all will be accepted.

**Reply by AntonReality** [reply]
So we want to show the preview from cameras without effects on preview but show effects on stream.
You pointed to your response where we can enqueue a raw CMSampleBuffer in MTKView. I’ve tried that but when not attaching stream to MTKView, the delegate method of the stream ‘didInput’ is not called at all, so we can’t get a CMSampleBuffer. 
This is reproducible in the example project as well, if you don’t attach the stream, the stream delegate methods are not called.
Thanks in advance!

**Reply by shogo4405** [reply]
Yes, that’s correct. That’s the design.The suggestion is to try attaching the stream to an unrelated MTHKView to the view you want to display.

**Reply by AntonReality** [reply]
Understood. Will try today. Thanks!

**Reply by AntonReality** [reply]
I tested this approach today. If using a dual camera it sends two raw buffers from the back and front cameras to MTKView. If you're not planning to add this functionality I'll work on a different ScreenObject that doesn't show effects on the preview.

Many thanks for considering my request.
